googler

```diff
@@ -1175,13 +1206,23 @@ class GoogleParser(html.parser.HTMLParser):
             self.set_handlers_to('result')
             return 'result'
 
+        # Autocorrect
+        if tag == 'span' and 'spell_orig' in self.classes(attrs) and attrs.get('id') != 'sifm':
+            self.autocorrected = True
+            return
+        if tag == 'a' and 'spell' in self.classes(attrs) and attrs.get('id') != 'srfl':
```

Comment:
> Doesn't this condition alone indicate auto-correction?

It doesn't. See parser docs:

```
    #   1. When npfr=1 (exact), there could still be an
    #      a.spell, in a block that reads (English version) "Did you mean:
    #      ...". Therefore, we only consider the query autocorrected when a
    #      meaningful .spell_orig is also present (self.autocorrected).
```

> we do around 7000 pattern matches and 800+ tag parsing for a page on an average.

I recall a benchmark earlier demonstrating that I/O is the real bottleneck.